### PR TITLE
Upgrade to 4.14 kernel

### PR DESCRIPTION
**Pull request by kraj**
Firmware and userland updates are needed for 4.14 to work.

### CODE REVIEW

**Comment by kraj** [comment]
I rebased it on top of pbarker/next

**Comment by betafive** [comment]
@kraj: Can the firmware & userland commits you've used still support a v4.9.y kernel? I want to keep both options working at least for the next few weeks while things transition.

**Comment by kraj** [comment]
@betafive I have not tested it with 4.9 however, we might have to use stable branch of firmware with 4.9, that probably means another recipe and may be another lockdown of PREFERRED_VERSION if someone choses to stick to 4.9

**Comment by betafive** [comment]
@kraj: The commits I used support both 4.9 and 4.14 kernels - I've boot tested both options. I've not done wifi & bluetooth testing as you have though.

I'd really like to have a point in time where both kernels work so that we can isolate any variation between them if we need to debug. I'm happy if things move to 4.14 only very soon after that though - we don't have the bandwidth to keep both kernels supported for long.

It also gives us one last update of firmware & kernel we can backport to rocko before master goes forward as 4.14 only.

**Comment by kraj** [comment]
I have looked into bootfiles recipe and from quick look it seems the portions we use from the firmware blob, we should be ok with 4.9 as well

File: recipes-kernel/linux/linux-raspberrypi_4.14.bb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 LINUX_VERSION ?= "4.14.22"
 
-SRCREV = "d95cdf3e6a0506e3a728b0c80ae02368779d2768"
```
Comment (agherzan) [inline recipes-kernel/linux/linux-raspberrypi_4.14.bb]: Is now an update and not an added recipe anymore.

**Comment by betafive** [comment]
I'm not too sure on the firmware. I had issues during the 4.4 -> 4.9 transition where 4.4 won't boot with more recent firmware. I'd like to stick to something that has been explicitly boot tested with 4.9 for at least a week or two to give people chance to test things out.

**Comment by agherzan** [comment]
@kraj Do you depend on the revision of firmware you update to in this PR? Would the one Paul has work for you too?

**Comment by kraj** [comment]
I think its better to use the newer one whichever that is. Since rpi org has been pushing 4.14 bugfixes to it. I have tested my set but I guess either would work.

**Comment by kraj** [comment]
I have rebased it again on top of master now

File: recipes-kernel/linux/linux-raspberrypi_4.14.bb (hunk 1 above)
Comment (kraj) [reply recipes-kernel/linux/linux-raspberrypi_4.14.bb]: yes this is an update on top of what was pushed as new recipe for 4.14

Comment (betafive) [reply recipes-kernel/linux/linux-raspberrypi_4.14.bb]: I think what Andrei means is that the commit message is now confusing.

**Comment by betafive** [comment]
The simplest way to fix this is for me to make some time to boot test this with a 4.9 kernel. I should be able to spare some time on Friday to do that. I'd rather be careful here so we don't disrupt other users unnecessarily.

General review comments:

* Commit message for linux-raspberrypi changes now need to say that it's an update not a new recipe.

* Changes to firmware.inc re-order lines un-necessarily. Can we tidy that up so it's more clear what has changed?

* Userland update looks good.

**Comment by kraj** [comment]
updated

**Comment by betafive** [comment]
Thanks @kraj, I'll test with 4.9 as soon as I get chance and then let's go ahead and merge this if all works.

**Comment by kraj** [comment]
@betafive any updates on this ?

**Comment by agherzan** [comment]
@betafive A friendly ping on this.

**Comment by betafive** [comment]
Was just about to comment on this. I've had a really busy couple of weeks, got some client work to finish off before Friday but then should have some time to look at this again. Apologies for the delay.

**Comment by betafive** [comment]
All looks good. Really sorry this took so long to test, had a few unexpected things come up at work.

**Comment by agherzan** [comment]
Better late... :) Thanks @betafive

**Pull request merged**